code/stl/Complex.hpp

```diff
@@ -3,72 +3,51 @@
 template <typename T=float>
 class Complex_t {
 public:
-    Complex_t(T r, T i);
-    Complex_t();
+    Complex_t() = default;
+    Complex_t(T r, T i) : m_r(r), m_i(i) {}
 
     T real() const {return m_r;};
     T imaginary() const {return m_i;};
 
-    Complex_t operator+(const Complex_t& other) const;
-    Complex_t operator-(const Complex_t& other) const;
-    Complex_t operator*(const Complex_t& other) const;
-    Complex_t operator*(const T factor) const;
-    Complex_t operator/(const T dividend) const;
-    Complex_t& operator+=(const Complex_t& other);
-    bool operator<(const Complex_t& a) const;
-private:
-    T m_r{}, m_i{};
-};
-
-typedef Complex_t<> Complex;
+    Complex_t& operator+=(const Complex_t& other) {
+        m_r += other.m_r;
+        m_i += other.m_i;
+        return *this;
+    }
 
-template <typename T>
-Complex_t<T>::Complex_t() {}
+    friend Complex_t operator+(Complex_t a, const Complex_t& b) {
+        return a += b;
+    }
 
-template <typename T>
-Complex_t<T>::Complex_t(T r, T i) : m_r(r), m_i(i) {}
+    friend Complex_t operator-(const Complex_t& a, const Complex_t& b) {
+        return Complex_t(a.m_r - b.m_r, a.m_i - b.m_i);
+    }
 
-template <typename T>
-Complex_t<T> Complex_t<T>::operator+(const Complex_t<T>& other) const {
-    return Complex_t(m_r + other.m_r, m_i + other.m_i);
-}
+    friend Complex_t operator*(const Complex_t& a, const Complex_t& b) {
+        return Complex_t(a.m_r*b.m_r - a.m_i*b.m_i,
+                         a.m_r*b.m_i + a.m_i*b.m_r);
+    }
 
-template <typename T>
-Complex_t<T> Complex_t<T>::operator-(const Complex_t<T>& other) const {
-    return Complex_t(m_r - other.m_r, m_i - other.m_i);
-}
+    friend Complex_t operator*(const Complex_t& c, T factor) {
+        return Complex_t(c.m_r*factor, c.m_i*factor);
+    }
```

Comment:
Here you removed `Complex*Complex`, or did I miss it? Is that intentional?

Reply:
Same for `/`

Reply:
We don't provide Complex/Complex, but that was the case also before ...

Reply:
New beef with the `operator*`:
It doesn't compile for `1.f * Complex{}`.

Reply:
I reproduced the feature set that was there before. But if it makes you feel better, I can add the missing operations :)

Reply:
I added the missing operators.